crates/h264/src/config.rs

```diff
@@ -91,29 +91,30 @@ pub struct AvccExtendedConfig {
     pub sequence_parameter_set_ext: Vec<SpsExtended>,
 }
 
-impl AVCDecoderConfigurationRecord {
-    /// Parses an AVCDecoderConfigurationRecord from a byte stream.
-    /// Returns a parsed AVCDecoderConfigurationRecord.
-    pub fn parse(reader: &mut io::Cursor<Bytes>) -> io::Result<Self> {
-        let configuration_version = reader.read_u8()?;
-        let profile_indication = reader.read_u8()?;
-        let profile_compatibility = reader.read_u8()?;
-        let level_indication = reader.read_u8()?;
-        let length_size_minus_one = reader.read_u8()? & 0b00000011;
-        let num_of_sequence_parameter_sets = reader.read_u8()? & 0b00011111;
+impl<'a> Deserialize<'a> for AVCDecoderConfigurationRecord<'a> {
+    fn deserialize<R>(mut reader: R) -> io::Result<Self>
+    where
+        R: scuffle_bytes_util::zero_copy::ZeroCopyReader<'a>,
+    {
+        let configuration_version = reader.as_std().read_u8()?;
```

Comment:
hmm, perhaps we can improve the ergonomics of the zero-copy reader a bit. I think one thing we can do is read it via the deserialize functions.